app/controllers/devise_registration_controller.rb

```diff
@@ -31,6 +31,7 @@ def start
     password_confirmation = params.dig(:user, :password_confirmation)
     password_length_ok = Devise.password_length.include? password&.length
     password_confirmation_ok = password == password_confirmation
+    password_denylist_ok = !BannedPassword.is_password_banned?(password)
```

Comment:
The list contains mixed case versions, suggest normalising and test lower case version only
```suggestion
    password_denylist_ok = !BannedPassword.is_password_banned?(password.downcase!)
```

Reply:
I've added the downcasing closer to where the database interaction happens---to `import_list` and `is_password_banned?`, rather than `import_from_ncsc` and the registration controller---to lessen the chances of this being broken in the future.